sympy/solvers/ode.py

```diff
@@ -5838,6 +5841,8 @@ def ode_lie_group(eq, func, order, match):
                 scoord = newcoord[-1]
                 try:
                     sol = solve([r - rcoord, s - scoord], x, y, dict=True)
+                    if sol == []:
+                        raise NotImplementedError
```

Comment:
I think continue rather than raise NotImplementedError would be better.

Reply:
The Lie group method works by trying out different heuristics. None of them is guaranteed to work for any particular ODE so the solver just tries them one after another. That's why we have a loop over the heuristics here. We need to make sure that if a particular heuristic doesn't work (which could be because solve returns an empty list or raises NotImplementedError) then the loop goes on to the next heuristic. Only when all of the heuristics has failed should we go on to raise NotImplementedError at the end.

The solve function will return an empty list if there are no solutions to the equation(s) or if the solution is infinite or possibly if there is an infinite number of solutions e.g.:
```julia
In [9]: solve(exp(x))                                                                                    
Out[9]: []

In [10]: solve([x-1, x-2], x)                                                                            
Out[10]: []
```

Reply:
I think that the "best" thing would be to factor the body of the loop out into a separate function and use `return None` rather than exceptions or continue.